### PR TITLE
Update to buildtools version 1.0.27-prerelease-01407-02, and enable creating transport packages

### PR DESCRIPTION
**Pull request by chcosta**
With this build tools update, we will be producing transport packages with the coreclr builds which include product and symbols.  Most of the changes are because buildtools has not been updated in coreclr for a while.

/cc @ericstj @weshaggard @noahfalk @ellismg

### CODE REVIEW

**Comment by weshaggard** [comment]
cc @wtgodbe who is also trying to update build tools in PR https://github.com/dotnet/coreclr/pull/9619

**Comment by chcosta** [comment]
@wtgodbe , let's sync up.

**Comment by wtgodbe** [comment]
@chcosta non-windows failures are looking like the following:

> 10:34:37        "/mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/build.proj" (Build target) (1) ->
10:34:37        "/mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/src/build.proj" (Build target) (2) ->
10:34:37        "/mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/src/mscorlib/facade/mscorlib.csproj" (Build target) (5) ->
10:34:37          /mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/Tools/Microsoft.Common.CurrentVersion.targets(1110,5): error MSB3644: The reference assemblies for framework ".NETPortable,Version=v4.5,Profile=Profile7" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend. [/mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/src/mscorlib/facade/mscorlib.csproj]
10:34:37          /mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/Tools/Microsoft.Common.CurrentVersion.targets(1110,5): error MSB3644: The reference assemblies for framework ".NETPortable,Version=v4.5" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend. [/mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/src/mscorlib/facade/mscorlib.csproj]
10:34:37 
10:34:37 
10:34:37        "/mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/build.proj" (Build target) (1) ->
10:34:37        "/mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/src/build.proj" (Build target) (2) ->
10:34:37        "/mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/src/mscorlib/ref/mscorlib.csproj" (Build target) (6) ->
10:34:37          /mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/Tools/Microsoft.Common.CurrentVersion.targets(1110,5): error MSB3644: The reference assemblies for framework ".NETFramework,Version=v4.0" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend. [/mnt/j/workspace/dotnet_coreclr/master/checked_ubuntu_prtest/src/mscorlib/ref/mscorlib.csproj]

I'm not sure where those target frameworks are coming from

**Comment by wtgodbe** [comment]
@dagood @mellinoe are either of you guys familiar with the above errors?

**Comment by wtgodbe** [comment]
I think the Windows pri-1 error is transient (OOM on a JIT test)

**Comment by mellinoe** [comment]
@dagood I think those errors are supposed to be suppressed somewhere in FrameworkTargeting.targets. Could you try using the version from corefx and see if it helps?

**Comment by chcosta** [comment]
It looks like there's a few things in corefx:Tools-Override\FrameworkTargeting.targets which wouldn't play nicely with coreclr.  Any idea which of the changes you're referring to?

**Comment by weshaggard** [comment]
I just looked to see why this doesn't happen in corefx and it is because of https://github.com/dotnet/corefx/blob/master/dir.targets#L41. I don't know why that would just now start being an issue but that should give you a workaround for now.

**Comment by mellinoe** [comment]
Yes, that looks like the logic I was thinking of.

**Comment by weshaggard** [comment]
Actually it looks like that used to be a warning but with the updated msbuild now it appears to be an error which is why it is now showing up.

**Comment by chcosta** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test please

File: src/mscorlib/System.Private.CoreLib.csproj [hunk 1]
```diff
@@ -35,6 +35,10 @@
     <SignAssembly>true</SignAssembly>
     <DelaySign>true</DelaySign>
     <DefineConstants>$(DefineConstants);CORECLR;_USE_NLS_PLUS_TABLE;RESOURCE_SATELLITE_CONFIG;CODE_ANALYSIS_BASELINE</DefineConstants>
+    <!-- We don't use any of MSBuild's resolution logic for resolving the framework, so just set these two properties to any folder that exists to skip
```
Comment (weshaggard) [inline src/mscorlib/System.Private.CoreLib.csproj]: Would you mind opening an issue in BuildTools about this and linking to it here in the comments. Thanks.

Comment (chcosta) [inline src/mscorlib/System.Private.CoreLib.csproj]: https://github.com/dotnet/buildtools/issues/1381

**Pull request merged**